mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -775,8 +778,49 @@ void dispatchShouldNotPersistEventWhenDispatchingWithKeysGetError() {
             }
 
             getSpeedProfile().longWaitCondition()
-                .untilAsserted(() ->
-                    assertThat(deadLetter().containEvents().block()).isFalse());
+                .untilAsserted(() -> assertThat(dispatchingFailureEvents()).containsOnly(EVENT));
+        }
+
+        @Test
+        void dispatchShouldPersistOnlyOneEventWhenDispatchingMultiGroupsGetError() {
+            EventCollector eventCollector = eventCollector();
+            eventBus().register(eventCollector, GROUP_A);
+            eventBus().register(eventCollector, GROUP_B);
+
+            rabbitMQExtension.getRabbitMQ().pause();
+
+            try {
+                eventBus().dispatch(EVENT, NO_KEYS).block();
+            } catch (Exception e) {
+                // ignore
+            }
+
+            getSpeedProfile().longWaitCondition()
+                .untilAsserted(() -> assertThat(dispatchingFailureEvents()).containsOnly(EVENT));
```

Comment:
The condition could succeed without verifying the expected behavior if other events arrived later.
But this allow "fast" tests and the more correct alternative would need to wait a given time and then verify that we have received only and only one event. 
So i am not sure what we should do on this one. Others opinions ?

Reply:
You're right, I was copy pasting in hurry! :)